signer/src/transaction_coordinator.rs

```diff
@@ -390,6 +392,75 @@ where
         .map(|_| ())
     }
 
+    /// Constructs a BitcoinPreSignRequest from the given transaction package and
+    /// sends it to the signers. Waits for acknowledgments from the signers until
+    /// the threshold is met or a timeout occurs.
+    /// If the signal stream closes unexpectedly, triggers a shutdown.
+    #[tracing::instrument(skip_all)]
+    async fn construct_and_send_bitcoin_presign_request(
+        &mut self,
+        bitcoin_chain_tip: &model::BitcoinBlockHash,
+        pending_requests: &SbtcRequests,
+        transaction_package: &[utxo::UnsignedTransaction<'_>],
+    ) -> Result<(), Error> {
+        // Create the BitcoinPreSignRequest from the transaction package
+        let sbtc_requests = BitcoinPreSignRequest {
+            request_package: transaction_package
+                .iter()
+                .map(|tx| (&tx.requests).into())
+                .collect(),
+            fee_rate: pending_requests.signer_state.fee_rate,
+            last_fees: pending_requests.signer_state.last_fees.map(Into::into),
+        };
+        let presign_ack_filter = {
+            let target_tip = *bitcoin_chain_tip;
+            move |event: &SignerSignal| {
+                matches!(
+                    event,
+                    SignerSignal::Event(SignerEvent::TxSigner(TxSignerEvent::MessageGenerated(
+                        Signed {
+                            inner: SignerMessage {
+                                bitcoin_chain_tip,
+                                payload: Payload::BitcoinPreSignAck(_),
+                            },
+                            ..
+                        },
+                    ))) if *bitcoin_chain_tip == target_tip
+                )
+            }
+        };
+
+        // Create a signal stream with the defined filter
+        let signal_stream = self.context.as_signal_stream(presign_ack_filter);
+
+        // Send the presign request message
+        self.send_message(sbtc_requests, bitcoin_chain_tip).await?;
+
+        tokio::pin!(signal_stream);
+        let future = async {
+            let mut num_acks = 1;
+            while num_acks < self.threshold {
+                // If signal_stream.next() returns None then one of the
+                // underlying streams has closed. That means either the
+                // network stream, the internal message stream, or the
+                // termination handler stream has closed. This is all bad,
+                // so we trigger a shutdown.
+                let Some(_) = signal_stream.next().await else {
+                    tracing::warn!("signal stream returned None, shutting down");
```

Comment:
I think it should suffice to just return an error here, the main loop will exit if it receives `None`.

Reply:
I wrote it this way because that's how we handle this error in `sign_stacks_transaction`, but I can change it if it's sure we don't need it! https://github.com/stacks-network/sbtc/blob/main/signer/src/transaction_coordinator.rs#L730-L733

Reply:
Switched to just returning an error in https://github.com/stacks-network/sbtc/pull/956/commits/fd4873d66ef4e4e368ae594b19bfec7e5f1bac1a